src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp

```diff
@@ -0,0 +1,277 @@
+    loop.exec();
+
+    while ( current_loading_action.getStatus() != QString::fromStdString( "LOADED") ) {
+        api->loadingActionIdGet( action_id );
+
+        QTimer::singleShot(5000, &loop, &QEventLoop::quit);
+        loop.exec();
+        // usleep coversion from seconds to microseconds
+        usleep( polling_frequency * 1e6 );
+    }
+
+} 
+
+void WebServiceClient::pollUnloadingAction( QString action_id) {
+    QEventLoop loop;
+
+     // Call back for Get /unloading/{action_id}
+    connect(api, &OAIDefaultApi::unloadingActionIdGetSignal, [&](OAIContainerActionStatus unloading_action) {
+        FILE_LOG(logERROR) << "Success /unloading/{action_id} GET : " << unloading_action.asJson().toStdString();
+        current_unloading_action = unloading_action; 
+        
+        loop.quit();
+    });
+    // Error call back for Get /unloading/{action_id}
+    connect(api, &OAIDefaultApi::unloadingActionIdGetSignalE, 
+        [&](OAIContainerActionStatus unloading_action , QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure unloading/{action_id} GET :" << error_str.toStdString();
+        FILE_LOG(logERROR) << error_code;
+
+        loop.quit();
+    });
+
+    api->unloadingActionIdGet( action_id );
+    QTimer::singleShot(0, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    while ( current_unloading_action.getStatus() != QString::fromStdString( "UNLOADED") ) {
+        
+        api->unloadingActionIdGet( action_id );
+        QTimer::singleShot(0, &loop, &QEventLoop::quit);
+        loop.exec();
+        
+        // usleep coversion from seconds to microseconds
+        usleep( polling_frequency * 1e6 );
+
+    }
+}
+
+int WebServiceClient::pollInspectionAction( QString action_id ) {
+    QEventLoop loop;
+
+    // Call back for GET /inspection/{action_id}
+    connect(api, &OAIDefaultApi::inspectionActionIdGetSignal, [&](OAIInspectionStatus inspection) {
+            current_inspection = inspection; 
+            FILE_LOG(logERROR) << "Success /inspection/{action_id} GET : " << current_inspection.asJson().toStdString() ;
+            loop.quit();
+    });
+    // Error call back for /inspection/{action_id}
+    connect(api, &OAIDefaultApi::inspectionActionIdGetSignalE, 
+        [&](OAIInspectionStatus inspection , QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure /inspection/{action_id} GET : " << error_str.toStdString();
+        FILE_LOG(logERROR) << "Failure /inspection/{action_id} GET : " << error_code;
+        loop.quit();
+    });
+
+    api->inspectionActionIdGet( action_id );
+    QTimer::singleShot(0, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    while ( true ) {
+
+        api->inspectionActionIdGet( action_id );
+        QTimer::singleShot(0, &loop, &QEventLoop::quit);
+        loop.exec();
+        
+        if (current_inspection.getStatus() == QString::fromStdString( "PASSED")){
```

Comment:
if none of this status match, what will this method return?

Reply:
I thought about checking against other possible values but there shouldn't be any other possible values in practice. I can make it return -1 in that case but that should never happen in practice. I will make the change for testing.